bin/elixir

```diff
@@ -56,63 +83,111 @@ while [ $I -le $# ]; do
         ;;
     -v|--compile|--no-halt)
         ;;
-    -e|-r|-pr|-pa|-pz|--remsh|--app)
+    -e|-r|-pr|-pa|-pz|--app|--eval|--remsh)
         S=2
         ;;
-    --detached|--hidden)
-        ERL="$ERL `echo $PEEK | cut -c 2-`"
+    --rpc-eval)
+        S=3
+        ;;
+    --detached)
+        echo "warning: the --detached option is deprecated" >&2
+        ERL="$ERL -detached"
+        ;;
+    --hidden)
+        ERL="$ERL -hidden"
         ;;
     --cookie)
-        I=$(expr $I + 1)
+        I=$(($I + 1))
         eval "VAL=\${$I}"
         ERL="$ERL -setcookie "$VAL""
         ;;
     --sname|--name)
-        I=$(expr $I + 1)
+        I=$(($I + 1))
         eval "VAL=\${$I}"
-        ERL="$ERL `echo $PEEK | cut -c 2-` "$VAL""
+        ERL="$ERL $(echo $PEEK | cut -c 2-) "$VAL""
         ;;
     --logger-otp-reports)
-        I=$(expr $I + 1)
+        I=$(($I + 1))
         eval "VAL=\${$I}"
         if [ "$VAL" = 'true' ] || [ "$VAL" = 'false' ]; then
-            ERL="$ERL -logger handle_otp_reports "$VAL""
+          ERL="$ERL -logger handle_otp_reports "$VAL""
         fi
         ;;
     --logger-sasl-reports)
-        I=$(expr $I + 1)
+        I=$(($I + 1))
         eval "VAL=\${$I}"
         if [ "$VAL" = 'true' ] || [ "$VAL" = 'false' ]; then
-            ERL="$ERL -logger handle_sasl_reports "$VAL""
+          ERL="$ERL -logger handle_sasl_reports "$VAL""
         fi
         ;;
     --erl)
-        I=$(expr $I + 1)
+        I=$(($I + 1))
         eval "VAL=\${$I}"
         ERL="$ERL "$VAL""
         ;;
+    --vm-args)
+        I=$(($I + 1))
+        eval "VAL=\${$I}"
+        ERL="$ERL -args_file "$VAL""
+        ;;
+    --boot)
+        I=$(($I + 1))
+        eval "VAL=\${$I}"
+        ERL="$ERL -boot "$VAL""
+        ;;
+    --boot-var)
+        I=$(($I + 1))
+        eval "KEY=\${$I}"
+        I=$(($I + 1))
+        eval "VAL=\${$I}"
+        ERL="$ERL -boot_var "$KEY" "$VAL""
+        ;;
+    --pipe-to)
+        I=$(($I + 1))
+        eval "RUN_ERL_PIPE=\${$I}"
+        if [[ "$RUN_ERL_PIPE" == "-"* ]]; then
+          echo "--pipe-to : PIPEDIR cannot be a switch" >&2
+          exit 1
+        fi
+
+        I=$(($I + 1))
+        eval "RUN_ERL_LOG=\${$I}"
+        if [[ "$RUN_ERL_LOG" == "-"* ]]; then
+          echo "--pipe-to : LOGDIR cannot be a switch" >&2
```

Comment:
```suggestion
          echo "--pipe-to : LOG_DIR cannot be a switch" >&2
```

Reply:
Erlang refers to those and other environment variables without snake case so I am going with their convention for now.